### DISCUSSION TITLE
Codemagic Build Stuck on Splash Screen

### DISCUSSION THREAD

**Discussion by yonycherkos** [codemagic-ci-cd/codemagic-docs#2872] [Q&A] [answered]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

Dear Codemagic Community,

We are currently building a Flutter app, and while the app works perfectly when run locally and has been successfully published to beta testers, we are encountering an issue with builds generated via Codemagic. Specifically, the build created with Codemagic gets stuck on the splash screen when launched on the simulator, while locally generated builds run without any issues.
The Codemagic-generated build gets stuck on the splash screen, preventing the integrated QA script from executing successfully. Notably, the QA script has been thoroughly tested locally and performs as expected. To further diagnose the issue, we installed the artifact generated by Codemagic on a local simulator and observed the same behavior—the app remains stuck on the splash screen.

We have checked for any configuration mismatches but have not been able to identify the root cause of this issue. The app works fine on the simulator when built locally and does not encounter this problem.
Could you kindly assist us in identifying any potential configuration or environment issues within the Codemagic workflow that might be causing this? Any guidance or troubleshooting steps would be greatly appreciated.

Thank you for your time and support. We look forward to your assistance.

Best regards,
Rejuve Team

### Expected results

The build should install and launch successfully on the emulator, and the QA script should execute without any issues.

### Actual results

The build was successfully installed on the emulator; however, the app failed to launch. Consequently, the QA script also failed, reporting an inability to locate the expected widgets.

### Build id (optional)

67514d81a6c8951abdeacec2

**Comment by dam-ease** [comment]
Hello @yonycherkos,
From your logs, I see the first screen seems to be a login page and also can see an error in the integration test step:
```
MaxRetryError: HTTPConnectionPool(host='localhost', port=4723): Max retries exceeded with url: /session (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x106199fa0>: Failed to establish a new connection: [Errno 61] Connection refused'))
```
Can you please share more information about the first screen of the app and what functionality it has?

From experience, it generally occurs due to missing credentials, e.g., Firebase config files or some conflict with the libs added and another lib that were already in the project. Can you check this out? Also, do you have any error handling going on in your app either with Sentry of Firebase Crashlytics?

Also, a user shared a solution: https://stackoverflow.com/questions/56407299/where-to-save-googleservice-info-plist-in-flutter-without-xcode that worked in his case

**Comment by tksuns12** [comment]
I have a same symptom Build number is `674fa1ae20bf0debefdea73f`.

It happens before `MyApp` is on, Sentry actually can't catch the problem.

**Reply by tksuns12** [reply]
On logcat it just repeats rendering splash screen and can't get into MainActivity. The thing is it happens on iOS app too.

<details>
<summary>Logs</summary>

```bash
12-06 19:15:09.810  1663  1804 D RenderEngine: [SEC_GC_CMN_SF_EFFECTS] drawLayersInternal,1279, Rendering layer Splash Screen kr.mathtutor.dev$_4178#24159 at index 1
12-06 19:15:10.817  1663  1804 D RenderEngine: [SEC_GC_CMN_SF_EFFECTS] drawLayersInternal,1279, Rendering layer Splash Screen kr.mathtutor.dev$_4178#24159 at index 1
12-06 19:15:12.838  1663  1804 D RenderEngine: [SEC_GC_CMN_SF_EFFECTS] drawLayersInternal,1279, Rendering layer Splash Screen kr.mathtutor.dev$_4178#24159 at index 1
12-06 19:15:13.836  1663  1804 D RenderEngine: [SEC_GC_CMN_SF_EFFECTS] drawLayersInternal,1279, Rendering layer Splash Screen kr.mathtutor.dev$_4178#24159 at index 1
12-06 19:15:14.815  1663  1804 D RenderEngine: [SEC_GC_CMN_SF_EFFECTS] drawLayersInternal,1279, Rendering layer Splash Screen kr.mathtutor.dev$_4178#24159 at index 1
12-06 19:15:17.847  1663  1804 D RenderEngine: [SEC_GC_CMN_SF_EFFECTS] drawLayersInternal,1279, Rendering layer Splash Screen kr.mathtutor.dev$_4178#24159 at index 1
12-06 19:15:22.687  1663  1804 D RenderEngine: [SEC_GC_CMN_SF_EFFECTS] drawLayersInternal,1279, Rendering layer Splash Screen kr.mathtutor.dev$_4178#24159 at index 1
12-06 19:15:22.900  1663  1804 D RenderEngine: [SEC_GC_CMN_SF_EFFECTS] drawLayersInternal,1279, Rendering layer Splash Screen kr.mathtutor.dev$_4178#24159 at index 1
12-06 19:15:23.111  1663  1804 D RenderEngine: [SEC_GC_CMN_SF_EFFECTS] drawLayersInternal,1279, Rendering layer Splash Screen kr.mathtutor.dev$_4178#24159 at index 1
12-06 19:15:23.212  1663  1804 D RenderEngine: [SEC_GC_CMN_SF_EFFECTS] drawLayersInternal,1279, Rendering layer Splash Screen kr.mathtutor.dev$_4178#24159 at index 1
12-06 19:15:24.888  1663  1804 D RenderEngine: [SEC_GC_CMN_SF_EFFECTS] drawLayersInternal,1279, Rendering layer Splash Screen kr.mathtutor.dev$_4178#24159 at index 1
12-06 19:15:25.891  1663  1804 D RenderEngine: [SEC_GC_CMN_SF_EFFECTS] drawLayersInternal,1279, Rendering layer Splash Screen kr.mathtutor.dev$_4178#24159 at index 1
12-06 19:15:34.516  1663  1804 D RenderEngine: [SEC_GC_CMN_SF_EFFECTS] drawLayersInternal,1279, Rendering layer Splash Screen kr.mathtutor.dev$_4178#24159 at index 1
12-06 19:15:38.938  1663  1804 D RenderEngine: [SEC_GC_CMN_SF_EFFECTS] drawLayersInternal,1279, Rendering layer Splash Screen kr.mathtutor.dev$_4178#24159 at index 1
12-06 19:15:53.578  1663  1804 D RenderEngine: [SEC_GC_CMN_SF_EFFECTS] drawLayersInternal,1279, Rendering layer Splash Screen kr.mathtutor.dev$_4178#24159 at index 1
12-06 19:15:53.785  1663  1804 D RenderEngine: [SEC_GC_CMN_SF_EFFECTS] drawLayersInternal,1279, Rendering layer Splash Screen kr.mathtutor.dev$_4178#24159 at index 1
12-06 19:15:53.885  1663  1804 D RenderEngine: [SEC_GC_CMN_SF_EFFECTS] drawLayersInternal,1279, Rendering layer Splash Screen kr.mathtutor.dev$_4178#24159 at index 1
12-06 19:16:00.052  1663  1804 D RenderEngine: [SEC_GC_CMN_SF_EFFECTS] drawLayersInternal,1279, Rendering layer Splash Screen kr.mathtutor.dev$_4178#24159 at index 1
12-06 19:16:03.779  1663  1804 D RenderEngine: [SEC_GC_CMN_SF_EFFECTS] drawLayersInternal,1279, Rendering layer Splash Screen kr.mathtutor.dev$_4178#24159 at index 1
12-06 19:16:03.991  1663  1804 D RenderEngine: [SEC_GC_CMN_SF_EFFECTS] drawLayersInternal,1279, Rendering layer Splash Screen kr.mathtutor.dev$_4178#24159 at index 1
12-06 19:16:04.091  1663  1804 D RenderEngine: [SEC_GC_CMN_SF_EFFECTS] drawLayersInternal,1279, Rendering layer Splash Screen kr.mathtutor.dev$_4178#24159 at index 1
12-06 19:16:09.017  1663  1804 D RenderEngine: [SEC_GC_CMN_SF_EFFECTS] drawLayersInternal,1279, Rendering layer Splash Screen kr.mathtutor.dev$_4178#24159 at index 1
12-06 19:16:12.942  1663  1804 D RenderEngine: [SEC_GC_CMN_SF_EFFECTS] drawLayersInternal,1279, Rendering layer Splash Screen kr.mathtutor.dev$_4178#24159 at index 1
12-06 19:16:13.047  1663  1804 D RenderEngine: [SEC_GC_CMN_SF_EFFECTS] drawLayersInternal,1279, Rendering layer Splash Screen kr.mathtutor.dev$_4178#24159 at index 1
12-06 19:16:13.148  1663  1804 D RenderEngine: [SEC_GC_CMN_SF_EFFECTS] drawLayersInternal,1279, Rendering layer Splash Screen kr.mathtutor.dev$_4178#24159 at index 1
12-06 19:16:13.249  1663  1804 D RenderEngine: [SEC_GC_CMN_SF_EFFECTS] drawLayersInternal,1279, Rendering layer Splash Screen kr.mathtutor.dev$_4178#24159 at index 1
12-06 19:16:15.113  1663  1804 D RenderEngine: [SEC_GC_CMN_SF_EFFECTS] drawLayersInternal,1279, Rendering layer Splash Screen kr.mathtutor.dev$_4178#24159 at index 1
12-06 19:16:15.213  1663  1804 D RenderEngine: [SEC_GC_CMN_SF_EFFECTS] drawLayersInternal,1279, Rendering layer Splash Screen kr.mathtutor.dev$_4178#24159 at index 1
12-06 19:16:23.180  1663  1804 D RenderEngine: [SEC_GC_CMN_SF_EFFECTS] drawLayersInternal,1279, Rendering layer Splash Screen kr.mathtutor.dev$_4178#24159 at index 1
1
```
</details>

**Reply by dam-ease** [reply]
Hello @tksuns12,
Thanks for sharing this!
Can you please share more information about the first screen of the app and what functionality it has? I see you are making use of firebase as well, there is a possibility there are missing firebase credentials in codemagic which results in a check on the first screen being blocked. Can you check out for your firebase credentials?

**Reply by tksuns12** [reply]
Actually I suspected one thing, which was that maybe an env variable has a typo because I don't know sometimes blank letter could be stuck in there. So I pasted a new value again ang it worked. It was totally my mistake. Thank you for help.

**Reply by dam-ease** [reply]
Glad this is resolved now!
@yonycherkos Please can you share if you have been able to check this out on your end as well?
Kindly let me know if you have any questions or concerns

**Reply by yonycherkos** [reply]
Thank you all for your responses. I successfully resolved the issue with the app being stuck on the splash screen by dynamically generating the .env file from the environment variable definitions.

However, we are encountering a related issue with the login and authentication flow. We are utilizing AWS Amplify for authentication and have double-checked that the environment variable definitions for the Cognito keys are correct. 

Any insights or suggestions regarding this issue would be greatly appreciated.

Thank you!

**Reply by dam-ease** [reply]
Glad to hear this has been resolved on your end as well.
You can checkout this section of the official docs on headless authentication - https://docs.amplify.aws/gen1/javascript/tools/cli/usage/headless/ to see if it helps. Also kindly share more details on the exact issues you are currently facing or any error message so I could tailor my response.

**Comment by yonycherkos** [comment]
I am currently experiencing an issue with the Amplify cognito API. When attempting to log in using a valid email and password, I receive a "something went wrong" error message from the API. This issue does not occur in my local environment or local build, where the login functionality works as expected.

The problem occurs only to the build generated through Codemagic. I have ensured that all necessary Cognito keys are included in the environment variables list and have verified that they contain the correct values.

I would appreciate any guidance or insights you may have regarding this matter. Thank you for your assistance.

**Reply by dam-ease** [reply]
Hello @yonycherkos,
From the error message, I think this might still be related to configuration keys. Can you please confirm that all the relevant keys are present in codemagic. Also, check in with the values that there are no hidden formatting issues (like extra spaces or line breaks). Also, is it possible you enable detailed logging for Amplify to get more context about the error during login? Or an error reporting or network inspection tool to get full logs on this?

**Reply by yonycherkos** [reply]
Hello @dam-ease
Thank for your reply. I did double check that the environment variable definition are correctly. To make sure there are no any error in the environment variable, I just define the variables directly in the codemagic.yaml file by copy and pasting it from local .env file. However, this still doesn't solve the login or authentication error.

**Reply by dam-ease** [reply]
Hello @yonycherkos
Thanks for your response!
Two other things come to mind now. How do you make use of the `.env ` file in your project, please can you share that. This is to confirm that your project gets access to these values. Also, I was checking out enabling [logging](https://docs.amplify.aws/gen1/android/build-a-backend/more-features/logging/set-up-logging/) to get more details on this error.
On another note, you have access to intercom, you can connect us via the Intercom widget from Codemagic Dashboard to get faster and dedicated responses.

**Comment by days155** [comment]
Hi!
I have the same issue.
IOS is working fine
Locally, the build is working, 
Debug build with Codemagic is working
But the release build with Codemagic stuck on the first screen right after the Splash screen. No errors or exceptions were caught by Sentry or Logcat.
Build number is 675c6d17d6a90b1632c5f8d4

**Reply by days155** [reply]
Migrating to Flutter's Gradle plugins helped to fixed the issue

**Comment by yonycherkos** [comment, marked as answer]
Hello,

I resolved the Flutter Amplify login and authentication error by removing the `--no-codesign` command from my build process. Initially, I was using the command:

`flutter build ios --simulator --no-codesign`

However, I found that the issue was resolved when I updated the command to:

`flutter build ios --simulator`

I am not entirely sure how the `--no-codesign` option affects Amplify authentication, especially since my Amplify configuration was correct. If anyone has insights on this matter, I would appreciate your input.

Thank you!
